test/test_spectral_ops.py

```diff
@@ -94,11 +94,37 @@ def _stft_reference(x, hop_length, window):
         X[:, m] = torch.fft.fft(slc * window)
     return X
 
-
 # Tests of functions related to Fourier analysis in the torch.fft namespace
 class TestFFT(TestCase):
     exact_dtype = True
 
+    # rocFFT requires/assumes that the input to hipfftExecC2R or hipfftExecZ2D
+    # is of the form that is a valid output from a real to complex transform
+    # (i.e. it cannot be a set of random numbers)
+    # So for ROCm, call np.fft.rfftn and use its output as the input
+    # for testing ops that call hipfftExecC2R
+    def _generate_valid_rocfft_input(self, input, op):
+        # check if op can invoke hipfftExecC2R or hipfftExecZ2D
+        if type(op) == SpectralFuncInfo:
+            supported_ops = op.supported_dtypes("")
+            if not all(ctype in supported_ops for ctype in [torch.cfloat, torch.double]):
+                return input
+        else:
+            if op.__name__ in ["fft_rfft2"]:
+                return input
+
+
+        # if input is complex use the real part
+        if torch.is_complex(input):
+            np_input_real = input.real.cpu().numpy()
+        else:
+            np_input_real = input.cpu().numpy()
+
+        # generate Hermitian symmetric input using rfftn
+        rfft_output = np.fft.rfftn(np_input_real)
+
+        return torch.from_numpy(rfft_output)
```

Comment:
@micmelesse `from_numpy` will always return a CPU tensor, right? So this never runs any tests on rocm.

Reply:
@peterbell10 I didn't notice that in the documentation. Hopefully, the op forces it to GPU. Either way, I will create a pr today to explicitly force the tensor to GPU.

Reply:
The ops do not force inputs to the correct device or dtype, they just call the corresponding torch function.

Reply:
This was was fixed in https://github.com/pytorch/pytorch/pull/61073